float32 width_mm
---
bool success
string message